src/sentry/plugins/migrator.py

```diff
@@ -25,37 +31,42 @@ def call(self):
                     # can disable the Plugin for all Projects. There'd be no
                     # Repos left, associated with the Plugin.
                     self.disable_for_all_projects(plugin)
+        self._logging_context()
 
-    def all_repos_migrated(self, provider):
+    def all_repos_migrated(self, provider: str) -> bool:
         return all(r.integration_id is not None for r in self.repos_for_provider(provider))
 
-    def disable_for_all_projects(self, plugin):
+    def disable_for_all_projects(self, plugin: Plugin2 | Plugin) -> None:
         for project in self.projects:
             try:
-                self.log(at="disable", project=project.slug, plugin=plugin.slug)
+                logger.info(
+                    "plugin.disabled", extra={"project": project.slug, "plugin": plugin.slug}
```

Comment:
If we did
```suggestion
                    "plugin.disabled", extra=self._logging_context({"project": project.slug, "plugin": plugin.slug})
```

And updated `_logging_context()` to accept a dictionary and return a dictionary we'd retain preserve the existing behavior more accurately.